docs/guide/clustering.md

```diff
@@ -15,32 +15,93 @@ is up to you and your performance needs.
 
-!!! note
+It is also recommended to use:
 
-    Memcached must not be bound to the local address if clusters are in use, or user logins will fail.
-    To avoid this, in `/etc/memcached.conf` take a look under `# Specify which IP address to listen on. The default is to listen on all IP addresses`
+- [Solr](solr.md) for better search and performance
+- a CDN for improved performance and faster ping time worldwide
+- active/passive database for failover
+- more recent versions of PHP and MySQL/MariaDB within [what is supported](../getting_started/requirements.md) for your eZ Platform version to get more performance out of each server. Numbers might vary so make sure to test this when upgrading.
 
-    For development environments, change the address below this comment in `/etc/memcached.conf` to `-l 0.0.0.0`
+### Shared persistence cache
 
-    For production environments, follow this more secure instruction from the Memcached man:
+[Redis](http://redis.io/), an in-memory data structure store, is the recommended cache solution for clustering.
+Redis is used via [Redis pecl extension](https://pecl.php.net/package/redis).
 
-    > -l &lt;addr&gt;
+An alternative cache solution for cluster setups is [using Memcached](persistence_cache.md#memcached).
 
-    > Listen on &lt;addr&gt;; default to INADDR\_ANY. &lt;addr&gt; may be specified as host:port. If you don't specify a port number, the value you specified with -p or -U is used. You may specify multiple addresses separated by comma or by using -l multiple times. This is an important option to consider as there is no other way to secure the installation. Binding to an internal or firewalled network interface is suggested.
+See [Redis Cache Adapter in Symfony documentation](https://symfony.com/doc/3.4/components/cache/adapters/redis_adapter.html#configure-the-connection)
+for information on how to configure Redis.
 
-It is also recommended to use:
+To use Redis, you need to set `cache_service_name` to `cache.redis`.
 
-- [Solr](solr.md) for better search and performance
-- a CDN for improved performance and faster ping time worldwide
-- active/passive database for failover
-- more recent versions of PHP and MySQL/MariaDB within [what is supported](../getting_started/requirements.md) for your eZ Platform version to get more performance out of each server. Numbers might vary so make sure to test this when upgrading.
+Example:
+
+``` yaml
+services:
+    cache.redis:
+        parent: cache.adapter.redis
+        tags:
+            - name: cache.pool
+              clearer: cache.app_clearer
+              provider: 'redis://secret@example.com:1234/13'
+```
+
+!!! caution "Clearing Redis cache"
+
+    The regular `php bin/console cache:clear` command does not clear Redis persistence cache.
+    To clear it, use a dedicated Symfony command: `php bin/console cache:pool:clear cache.redis`.
+
+##### Redis Clustering
```

Comment:
This can make sense to have here, but then we should rewrite it to more clearly cover both sessions and persistence cache topics. and link to it